nltk/translate/chrf_score.py

```diff
@@ -110,28 +132,59 @@ def corpus_chrf(list_of_references, hypotheses, min_len=1, max_len=6, beta=3.0):
     :type max_len: int
     :param beta: the parameter to assign more importance to recall over precision
     :type beta: float
+    :param ignore_whitespace: ignore whitespace characters in scoring
+    :type ignore_whitespace: bool
     :return: the sentence level CHRF score.
     :rtype: float
     """
 
-    assert len(list_of_references) == len(hypotheses), "The number of hypotheses and their references should be the same"
+    assert len(references) == len(hypotheses), (
+        "The number of hypotheses and their references should be the same")
+
+    # Keep f-scores for each n-gram order separate
+    n_gram_fscores = {k: [] for k in range(min_len, max_len + 1)}
+
+    num_sents = 0  # we don't call len() in case these are generators
 
     # Iterate through each hypothesis and their corresponding references.
-    for reference, hypothesis in zip(list_of_references, hypotheses):
-        # Cheating condition to allow users to input strings instead of tokens.
-        if type(reference) and type(hypothesis) != str:
-            reference, hypothesis = ' '.join(reference), ' '.join(hypothesis)
-        # For each order of ngram, calculate the no. of ngram matches and
-        # keep track of no. of ngram in references.
-        ref_ngrams = Counter(everygrams(reference, min_len, max_len))
-        hyp_ngrams = Counter(everygrams(hypothesis, min_len, max_len))
-        overlap_ngrams = ref_ngrams & hyp_ngrams
-        tp = sum(overlap_ngrams.values()) # True positives.
-        tpfp = sum(hyp_ngrams.values()) # True positives + False positives.
-        tffn = sum(ref_ngrams.values()) # True posities + False negatives.
-
-    precision = tp / tpfp
-    recall = tp / tffn
-    factor = beta**2
-    score = (1+ factor ) * (precision * recall) / ( factor * precision + recall)
-    return score
+    for reference, hypothesis in zip(references, hypotheses):
+        num_sents += 1
```

Comment:
This is not necessary, we can use the `enumerate()`.

Reply:
The value `num_sents` is used after the `for` loop finishes. Just to be sure, is this what you have in mind?

```
for num_sents, (reference, hypothesis) in zip(references, hypotheses):
    ...

num_sents += 1 # num_sents counts from zero
```

Reply:
`enumerate(..., start=1)` is an option. https://docs.python.org/3/library/functions.html#enumerate ;P

Reply:
I would argue that's less readable than the current version but I'm fine doing it this way.

Reply:
How about `num_sents = len(hypotheses` =)